public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -392,6 +404,26 @@ export class PanelModel {
     this.transformations = transformations;
     this.getQueryRunner().setTransformations(transformations);
   }
+
+  updateQueryRunnerFieldOverrides() {
+    if (!this.plugin) {
+      return;
+    }
+
+    this.getQueryRunner().setFieldOverrides({
+      fieldOptions: this.options.fieldOptions,
+      replaceVariables: this.replaceVariables,
+      custom: this.plugin.customFieldConfigs,
+      theme: config.theme,
+    });
+
+    this.getQueryRunner()
+      .getData()
+      .pipe(take(1))
+      .subscribe(val => {
+        this.getQueryRunner().pipeDataToSubject(val);
+      });
```

Comment:
Not sure this is the best way to make sure subscribers are notified. Any other ideas @torkelo or maybe @hugohaggmark ?

Reply:
I think I need a bit more context to help out with this

Reply:
@hugohaggmark here we go. When the field overrides are updated(options basically in the PanelModel) we need to update those in the PanelQueryRunner as well, as the data processing(applying the overrides and transformations) happen in PQR now. The processing is implemented in the getData method of PQR. The above code is a way of making sure that the data is processed after the update and then piped to the subject for components to update their state. This snippet doesn't look like something legitimate to me, but I'm not RxJS experts, so who knows?:)

Reply:
We already have the last result so should this not be:
```typescript
const panelQueryRunner = this.getPanelQueryRunner();
panelQueryRunner.pipeDataToSubject(panelQueryRunner.getLastResult()); 
```

Unrelated cleanup that I found confusing: PanelQueryRunner has pipeToSubject and pipeDataToSubject that do very different things, pipeDataToSubject should be renamed to maybe emitData or something (as it does not pipe anything in the rxjs sense)

Reply:
@torkelo but last result is only updated when the query is run. So it won't do it's job. It also does not have processed data, only preprocessed for further transformation and overrides application. I guess the `getData` should also update the `lastResult`

Reply:
Not sure I understand then. As the processing is done in getData (that listens to the subject)  and lastResult is stored pre processing so that should be fine

Reply:
@torkelo that's updated now

Reply:
ah wait, this needs some more love as data ain't updated when exiting edit mode